### DISCUSSION TITLE
udp_connection not fund issue at runtime

### DISCUSSION THREAD

**Discussion by lili1218** [contiki-ng/contiki-ng#1566] [Q&A]
Hi,
I  try to send multicast message. initially it works properly, node are able to receive multicast message, but after running for a while,it will have below error message

> 2021-03-31 07:24:36.498869 : [ERR : IPv6      ] udp: no matching connection found

after I put more debug msg, I found the register udp socket for remote IP is being changed( it should be all 0)

>  local socket info: 
uip_ntohs(uip_udp_conn->lport)=3001 , 
uip_ntohs(uip_udp_conn->rport)=3001, 
uip_udp_conn->ripaddr =  8400::
sender socket info: 
uip_ntohs(UIP_UDP_BUF->srcport)=3001,
uip_ntohs(UIP_UDP_BUF->destport)=3001,
UIP_IP_BUF->srcipaddr= fe80::212:4b00:2089:aaaa,
UIP_IP_BUF->destipaddr= ff1e::89:abcd

the code related to multicast program as below:
1. add multicast address and open socket
```
static uip_ds6_maddr_t *
join_mcast_group(void)
{
  uip_ds6_maddr_t *rv;
  const uip_ipaddr_t *default_prefix = uip_ds6_default_prefix();

  uip_ip6addr_copy(&bc_addr, default_prefix);
  uip_ds6_set_addr_iid(&bc_addr, &uip_lladdr);
  uip_ds6_addr_add(&bc_addr, 0, ADDR_AUTOCONF);

  uip_ip6addr(&bc_addr, 0xFF1E,0,0,0,0,0,0x89,0xABCD);
  //add myself to a multicast group
  rv = uip_ds6_maddr_add(&bc_addr);

  return rv;
}


PROCESS_THREAD(mcast_process, ev, data)
{
  if(join_mcast_group() == NULL) {
    printf("MC: Failed to join group\n");
    PROCESS_EXIT();
  }

  //set listening remote address = 0 add local/remote port = MCAST_SINK_UDP_PORT
  if(0 == simple_udp_register(&sink_conn, MCAST_SINK_UDP_PORT, NULL, MCAST_SINK_UDP_PORT,
                      multicast_rx_callback) ) {
    printf("MC: open sock failed\r\n");
    PROCESS_EXIT();
  }
.....
}
```
2. when receive multicast packet, send it out again
`simple_udp_sendto(&sink_conn, &buf, buf_pkt_len, &bc_addr);`

could someone share why this happens? something wrong with my coding or memory issue ?
thanks~

**Comment by atiselsts** [comment]
> I found the register udp socket for remote IP is being changed( it should be all 0)

This sounds like a memory corruption issue. Perhaps some other code is overrunning some buffers? Perhaps even related to your LINKADDR_SIZE change.
